File: 2023/advent_of_code/test/q14/q14_test.exs
defmodule AdventOfCode.Q14Test do
  use ExUnit.Case
  doctest AdventOfCode.Q14

  test "Q14 Part 1 Example works" do
    file_stream = File.stream!("./test/q14/q14_example.txt")
    result = AdventOfCode.Q14.part1(file_stream)
    assert result == 136
  end

  test "Q14 Part 1 works" do
    file_stream = File.stream!("./test/q14/q14.txt")
    result = AdventOfCode.Q14.part1(file_stream)
    assert result == 110_779
  end

  test "Q14 Part 2 Example works" do
    file_stream = File.stream!("./test/q14/q14_example.txt")
    result = AdventOfCode.Q14.part2(file_stream)
    assert result == 64
  end

  test "Q14 Part 2 works" do
    file_stream = File.stream!("./test/q14/q14.txt")
    result = AdventOfCode.Q14.part2(file_stream)
    assert result == 86069
  end
end

File: 2022/AdventOfCode/src/test/kotlin/q14/Q14Test.kt
package q14

import java.nio.file.Paths
import kotlin.test.Test
import kotlin.test.assertEquals

class Q14Test {
    @Test
    fun `part 1 works`() {
        val filePath = javaClass.getResource("/q14.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(779, part1(filePath!!))
    }

    @Test
    fun `part 2 works`() {
        val filePath = javaClass.getResource("/q14.txt")?.toURI()?.let { Paths.get(it) }
        assertEquals(27426, part2(filePath!!))
    }
}